### ISSUE TITLE
enhancement: Possibility to use SolidJS?

### ISSUE THREAD

**Issue by FlurryNight** [jrestall/remix-kit#8]
### Description

Just found about this solution and first of all, Thank you for doing it. Projects like this matter.

I was wondering if we could use **SolidJS** as the view library since u mentioned agnostic, I don't know if it's doable with **Remix** internals but **SolidJS** in **Remix** would be the cherry on top of the cake, at least until Solid Start reaches stable and gets mature.

Thank you everyone involved in this project and **Happy Holidays**🎅 

### Suggested solution

Provide a `@remix-kit/solidjs` package

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/jrestall/remix-kit/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by cliffordfajardo** [comment]
That would be nice ; I think one of the first things that might need to happen is getting an implementation of `react-router-dom` done in SolidJS

The remix core team posted this repo of work some people in the community are doing to create react-router-dom (i think eventually remix team will cause this `remix-router`  in other frameworks 
- https://github.com/brophdawg11/remix-routers/tree/main

One thing that's highlighted in the `remix-routers` repo above that SSR support in these implementations is work in progress. I think in a few months when remix team finishes integrating remix-router into remix core that might make the work simpler, lots of moving parts in the core remix repo happening right now
- https://remix.run/blog/react-routering-remix

**Comment by jrestall** [comment]
Hi @FlurryNight, appreciate the input and suggestion! 

As @cliffordfajardo mentioned it will depend on remix-routers support. We would then look to combine that with an existing Vite plugin to provide SSR support. Similar to how the @remix-kit/vue library is being setup (https://github.com/jrestall/remix-kit/blob/main/packages/remix-vue/src/setup.ts). 

remix-routers is only focussed on CSR though, so there is also a dependent piece of work to create a headless version of a lot of the @remix-run/react components like Links and RemixBrowser so that any view library (vue, preact, svelte, solid etc) could easily build their own components. Hopefully this would be done and maintained by Remix but if Vue or some other library is important to someone then they could make a start and potentially contribute it back to Remix. 

It's mentioned that remix-kit is view agnostic since none of the main libraries like cli, kit, vite care what view library you use and support can be added with a separate package e.g. @remix-kit/react.

**Comment by jrestall** [comment]
Progress is being made on a solid js router for Remix https://twitter.com/nivekithanS/status/1607019446308335616 👀

**Comment by cliffordfajardo** [comment]
@FlurryNight - looks like @nivekithan is working on a solidjs port of react router 👀 
- https://twitter.com/nivekithanS/status/1607754034185920513

**Comment by nivekithan** [comment]
Yeah I am working on solid js integration. Right now it's just integration with remix_run/router and that too only for client side. 

I have also fallen little behind on my pr (got an laptop issue) hopefully I will make an pr in next two days.

**Comment by nivekithan** [comment]
If the pr is accepted and the remix team is also interested in adding support for ssr, then yeah I am planning to work on it.
